storage/storage_test.go

```diff
@@ -509,6 +510,341 @@ func TestWriteRead(t *testing.T) {
 	}
 }
 
+func TestWriteReadMulti(t *testing.T) {
```

Comment:
(Non-blocking: A fair amount of this could be extracted into helpers (e.g. ”verify signatures match expected data”) shared between this and pre-existing tests.